src/utils.rs

```diff
@@ -0,0 +1,163 @@
+use crate::{Result, Size};
+use std::cmp;
+
+#[cfg(feature = "image")]
+use fast_image_resize as fr;
+#[cfg(feature = "image")]
+use {crate::errors::ImageError, image::RgbImage};
+
+// Get the appropriate level for the given dimensions: i.e. the level with at least one
+// dimensions (i.e along one axis) greater than the dimension requested
+#[cfg(feature = "image")]
+pub fn get_best_level_for_dimensions(
+    dimension: &Size,
+    dimension_level_0: &Size,
+    level_count: u32,
+) -> u32 {
+    let downsample = f64::max(
+        f64::from(dimension_level_0.w) / f64::from(dimension.w),
+        f64::from(dimension_level_0.h) / f64::from(dimension.h),
+    );
+    let level_dowsamples: Vec<f64> = (0..level_count)
```

Comment:
Here is a more Rustish way to achieve that https://github.com/AzHicham/bioformats-rs/blob/714882e1b14a9fb2f4d1c2b5a6a3a6c4f59e7e3d/src/extra/wsi/slide.rs#L98
(0..).map().take_while().enumerate().rfind().unwrap_or() 
I’m not sure but it should works ^^

Reply:
In biofirmats i think the code is broken we need to check against the first value (max slide dimension ie downsample of 1) then unwraps with the last value :/

Reply:
the implementation looks correct to me !

Reply:
I used the one from bioformats